packages/toggle/addon/styles/components/_nucleus-toggle.scss

```diff
@@ -0,0 +1,144 @@
+@import "nucleus/variables";
+
+$toggle-on: #2c5cc5;
+$toggle-off: #576c7d;
+
+$large-variant: 24px;
+$medium-variant: 20px;
+$small-variant: 16px;
+
+$large-thumb-size: calc(#{$large-variant} + 4px);
+$medium-thumb-size: calc(#{$medium-variant} + 4px);
+$small-thumb-size: calc(#{$small-variant} + 4px);
+
+$transform: calc(100% - 8px);
+
+.nucleus-toggle {
+  position: relative;
+  display: inline-block;
+
+  &--large {
+    width: 48px;
+    height: 32px;
+
+    .thumb {
+      width: $large-thumb-size;
+      height: $large-thumb-size;
+    }
+  }
+
+  &--medium {
+    width: 40px;
+    height: 28px;
+
+    .thumb {
+      width: $medium-thumb-size;
+      height: $medium-thumb-size;
+    }
+  }
+
+  &--small {
+    width: 32px;
+    height: 24px;
+
+    .thumb {
+      width: $small-thumb-size;
+      height: $small-thumb-size;
+    }
+  }
+
+  .track {
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    margin: 6px 2px;
+    border-radius: 12px;
+    background: $toggle-off;
+  }
+
+  .thumb {
+    top: 2px;
+    position: absolute;
+    transition: .3s;
+    -webkit-transition: .3s;
+    pointer-events: none;
+
+    &::before {
+      content: '';
+      opacity: 0;
+      width: inherit;
+      height: inherit;
+      background-color: $toggle-off;
+      position: absolute;
+      border-radius: 50%;
+    }
+
+    &__control {
+      top: 0;
+      left: 0;
+      right: 0;
+      bottom: 0;
+      margin: 2px;
+      position: absolute;
+      border-radius: 50%;
+      border: 1px solid $toggle-off;
+      background-color: $color-milk;
+      box-shadow: 0 1px 2px 0 rgba(39, 49, 58, .3);
+
+      &--icon {
+        position: absolute;
+        width: 8px;
+        height: 8px;
+        margin: auto;
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+      }
+    }
+  }
+
+  &[checked] {
+    .track {
+      background-color: $toggle-on;
+    }
+
+    .thumb {
+      transform: translateX($transform);
+      -ms-transform: translateX($transform);
+      -webkit-transform: translateX($transform);
+
+      &::before {
+        background-color: $toggle-on;
+      }
+
+      &__control {
+        border: 1px solid $toggle-on;
+      }
+    }
+  }
+
+  &:hover {
+    .thumb::before {
+      opacity: .4;
+    }
+  }
+
+  input {
+    width: inherit;
+    height: inherit;
```

Comment:
Is this inheritance necessary here?

Reply:
Yes..
Inherited parent's width & height, as input is an inline-block element. 
The control[Hover + focus + click] is on the input element not the wrapper element, as I'm not using <label>.